### PR TITLE
שליחת OTP מיידית דרך Celery + fallback לוואטסאפ

### PR DESCRIPTION
**Pull request by amirbiron**
## תיאור קצר
הוספת שליחה מיידית של OTP דרך Celery task (`send_message.delay`) במקום להמתין ל-beat scheduler. הוספת fallback ממודע מטלגרם לוואטסאפ כשאין `telegram_chat_id`, עם הגנה מפני שליחה כפולה בעיבוד מקביל.

## שינויים עיקריים
- [x] קוד (Backend / Services)
- [x] Celery / Workers
- [x] בדיקות

**פירוט:**
- **`app/api/routes/panel/auth.py`**: 
  - שליחה מיידית של OTP דרך `send_message.delay(msg.id)` בעת בקשת OTP
  - fallback ממודע מטלגרם לוואטסאפ כשאין `telegram_chat_id` (עם warning בלוג)
  - טיפול בשגיאות Celery עם fallback ל-beat scheduler
  - הסתרת מספרי טלפון בלוגים

- **`app/workers/tasks.py`**:
  - בדיקה אטומית בתחילת `_process_single_message` — דילוג על הודעות שכבר בעיבוד (מונע שליחה כפולה)
  - הוספת logging מפורט בעת שליחה מוצלחת/כושלת (עם הסתרת מספרי טלפון)

- **`tests/test_panel_auth.py`**:
  - `TestOTPDelivery` — 3 בדיקות חדשות:
    - `test_otp_triggers_immediate_celery_task`: אימות שקריאה ל-`send_message.delay` מתבצעת מיד
    - `test_otp_sent_via_telegram_when_chat_id_exists`: אימות שליחה דרך טלגרם כשיש `chat_id`
    - `test_otp_falls_back_to_whatsapp_without_chat_id`: אימות fallback לוואטסאפ ללא `chat_id`

## בדיקות
- [x] Unit tests — עוברים (3 בדיקות חדשות בקלאס `TestOTPDelivery`)
- [x] כיסוי ל-edge cases: fallback, Celery unavailable, duplicate processing

## CI Checks
- pytest (כל הבדיקות עוברות)

## סוג שינוי
- [x] feat: פיצ'ר חדש (שליחה מיידית + fallback)
- [x] fix: תיקון (הגנה מפני שליחה כפולה)

## צ'קליסט
- [x] כל הפלט בעברית (כותרות, הודעות לוג, הערות בקוד)
- [x] אין `print()` — רק `logger`
- [x] מספרי טלפון מוסתרים בלוגים (`PhoneNumberValidator.mask()`)
- [x] כל קלט עובר ולידציה (קיים בקוד הקודם)
- [x] type hints בפונקציות חדשות
- [x] בדיקות לפיצ'רים חדשים
- [x] אין `except Exception: pass` — טיפול מפורש בשגיאות
- [

https://claude.ai/code/session_01K4iTrhMWa12gEAZYZnc96E

### CODE REVIEW

**Comment by amirbiron** [comment]
הנה סיכום התיקון:

## מה נמצא

שלוש בעיות גרמו לכך שקוד האימות לא הגיע:

### 1. לא היה dispatch מיידי
הודעת ה-OTP נכנסה לטבלת outbox וחיכתה ל-Celery beat scheduler שרץ כל 10 שניות. אם היה backlog של 50+ הודעות (broadcast לשליחים למשל), ה-OTP תקע בסוף התור.

**תיקון:** `send_message.delay(msg.id)` מיד אחרי commit — שליחה מיידית דרך Celery task ייעודי.

### 2. אפס לוגים על שליחה
כשה-worker שלח הודעה ישירה (לא broadcast) בהצלחה — לא היה **שום** log. גם כשלון לא קיבל log ייעודי. לכן "אין כלום בלוגים".

**תיקון:** לוג מפורש עם `message_id`, `message_type`, `platform` בהצלחה ובכשלון (`app/workers/tasks.py`).

### 3. fallback שקט לפלטפורמה לא נכונה
כשבעל תחנה רשום כ-`platform="telegram"` אבל אין לו `telegram_chat_id` (כי הוספו דרך מספר טלפון ומעולם לא פתחו את הבוט), הקוד שלח את ה-OTP דרך WhatsApp **בלי שום warning**. אם אין להם WhatsApp — ההודעה נעלמה.

**תיקון:** `logger.warning` מפורש כשיש fallback לפלטפורמה אחרת, + guard נגד מצב שאין שום אמצעי שליחה.

### הגנה נוספת
Guard ב-`_process_single_message` שבודק שההודעה עדיין PENDING לפני עיבוד — מונע שליחה כפולה כשגם ה-delay task וגם ה-beat רצים במקביל.

File: app/workers/tasks.py [hunk 1]
```diff
@@ -182,8 +182,20 @@ async def _process_single_message(message: OutboxMessage) -> tuple:
...
+            sa_update(OutboxMessage)
+            .where(
+                OutboxMessage.id == message.id,
+                OutboxMessage.status == MessageStatus.PENDING,
+            )
+            .values(status=MessageStatus.PROCESSING)
+        )
+        await db.commit()
+        if claim_result.rowcount == 0:
+            return True, "Already processed or in progress"
```
[the remark this answers is not in the document]
Comment (amirbiron) [reply app/workers/tasks.py]: בעתיד
